hackathon_site/dashboard/frontend/src/slices/event/teamAdminSlice.ts

```diff
@@ -0,0 +1,98 @@
+import {
+    createAsyncThunk,
+    createEntityAdapter,
+    createSelector,
+    createSlice,
+} from "@reduxjs/toolkit";
+import { get } from "api/api";
+import { APIListResponse, Team } from "api/types";
+import { AppDispatch, RootState } from "slices/store";
+import { displaySnackbar } from "slices/ui/uiSlice";
+
+interface TeamAdminExtraState {
+    errorState: string | null;
+    isMoreLoading: boolean;
+    isLoading: boolean;
+}
+
+interface RejectValue {
+    status: number;
+    message: any;
+}
+
+const extraState: TeamAdminExtraState = {
+    errorState: null,
+    isLoading: false,
+    isMoreLoading: false,
+};
+
+export const teamAdminReducerName = "adminTeam";
+const teamAdminAdapter = createEntityAdapter<Team>();
+export const initialState = teamAdminAdapter.getInitialState(extraState);
+
+// Thunks
+
+export const getAllTeams = createAsyncThunk<
+    APIListResponse<Team>,
+    void,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(`${teamAdminReducerName}/getAllTeams`, async (_, { dispatch, rejectWithValue }) => {
+    try {
+        const response = await get<APIListResponse<Team>>("/api/event/teams/", {
+            limit: 24,
```

Comment:
could you put `24` as a constant somewhere at the top of this file

Reply:
alright